File: java/src/test/java/io/binac/leetcode/SubsetsTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class SubsetsTest {
    private final Subsets.Solution1 SOLUTION1 = new Subsets.Solution1();

    private final int[] NUMS1 = {1, 2, 3};
    private final List<List<Integer>> EXPECTED1 = Arrays.asList(
            Collections.emptyList(),
            Collections.singletonList(1),
            Arrays.asList(1, 2),
            Arrays.asList(1, 2, 3),
            Arrays.asList(1, 3),
            Collections.singletonList(2),
            Arrays.asList(2, 3),
            Collections.singletonList(3)
    );

    private final int[] NUMS2 = {0};
    private final List<List<Integer>> EXPECTED2 = Arrays.asList(
            Collections.emptyList(),
            Collections.singletonList(0)
    );

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.subsets(NUMS1));
        assertEquals(EXPECTED2, SOLUTION1.subsets(NUMS2));
    }
}

File: cpp/tests/SubsetsTest.cpp
#include "Subsets.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(Subsets, Solution78_1) {
  auto S78_1 = Solution78_1();

  std::vector<int> Nums1{1, 2, 3};
  std::vector<std::vector<int>> const Expected1{
      {}, {1}, {1, 2}, {1, 2, 3}, {1, 3}, {2}, {2, 3}, {3},
  };
  EXPECT_EQ(Expected1, S78_1.subsets(Nums1));

  std::vector<int> Nums2{0};
  std::vector<std::vector<int>> const Expected2{{}, {0}};
  EXPECT_EQ(Expected2, S78_1.subsets(Nums2));
}